buildinfo/buildinfo.go

```diff
@@ -48,6 +54,34 @@ func Version() string {
 	return version
 }
 
+// VersionsMatch compares the two versions. It assumes the versions match if
+// the major and the minor versions are equivalent. Patch versions are
+// disregarded. If it detects that either version is a developer build it
+// returns true.
+func VersionsMatch(v1, v2 string) bool {
+	// Developer versions are disregarded...hopefully they know what they are
+	// doing.
+	if strings.HasPrefix(v1, develPrefix) || strings.HasPrefix(v2, develPrefix) {
+		return true
+	}
+
+	v1Toks := strings.Split(v1, ".")
+	v2Toks := strings.Split(v2, ".")
+
+	// Versions should be formatted as "<major>.<minor>.<patch>".
+	// We assume malformed versions are evidence of a bug and return false.
+	if len(v1Toks) < 3 || len(v2Toks) < 3 {
+		return false
+	}
+
+	// Slice off the patch suffix. Patch versions should be non-breaking
+	// changes.
+	v1MajorMinor := strings.Join(v1Toks[:2], ".")
+	v2MajorMinor := strings.Join(v2Toks[:2], ".")
+
+	return v1MajorMinor == v2MajorMinor
```

Comment:
You should use `semver` to compare instead, that way you don't have to manually parse any of this.

Reply:
https://pkg.go.dev/golang.org/x/mod/semver

Reply:
Ngl that package sucks

Reply:
Meh I think it's better than doing it ourselves

Reply:
I'll give it a go because i <3 you but last time I tried to use it, it disappointed me

Reply:
actually worked out pretty well for this

Reply:
i take that back it messes everything up.